libvast/vast/view.hpp

```diff
@@ -230,8 +250,8 @@ class container_view_iterator
     --position_;
   }
 
-  template <class T>
-  void advance(T n) {
+  template <class N>
```

Comment:
As far as I can tell, this `iterator` models a random-access iterator. Why use special member functions instead of sticking to [the STL concept RandomAccessIterator](http://en.cppreference.com/w/cpp/concept/RandomAccessIterator) by providing `+`, `+=`, etc. operators?

Reply:
Because I'm relying on `detail::iterator_facade` to fill in exactly those details. It's non-trivial to manually generate all necessary members of a full random-access iterator. For details, see the [Boost docs](https://www.boost.org/doc/libs/1_67_0/libs/iterator/doc/iterator_facade.html).